src/panels/lovelace/cards/hui-grid-card.ts

```diff
@@ -87,10 +87,14 @@ class HuiGridCard extends HuiStackCard<GridCardConfig> {
           grid-column: 1 / 1;
         }
 
-        :host([square]) #root > *:first-child {
+        :host([square]) #root > *:not([invisible]) {
```

Comment:
Why don't we just check if the first child has the attribute `invisible?`

Reply:
If any card is visible, at least one needs `grid-area: 1/1/1/1`. So if the first card is invisible, the second one should take that spot, and so on.
And unfortunately `:first-of-type` doesn't work with attribute selectors.

Reply:
Oh... right it is because of the `::before` that targets `1/1/1/1`...